data/trie/interceptedNode.go

```diff
@@ -79,9 +78,14 @@ func (inTn *InterceptedTrieNode) IsInterfaceNil() bool {
 	return inTn == nil
 }
 
-// EncodedNode returns the intercepted encoded node
-func (inTn *InterceptedTrieNode) EncodedNode() []byte {
-	return inTn.SerializedNode.NodeBytes
+// GetSerialized returns the intercepted encoded node
+func (inTn *InterceptedTrieNode) GetSerialized() []byte {
+	return inTn.SerializedNode
+}
+
+// SetSerialized sets the given bytes as the SerializedNode
+func (inTn *InterceptedTrieNode) SetSerialized(serializedNode []byte) {
+	inTn.SerializedNode = serializedNode
```

Comment:
Pleas add a mutex protection to this. All intercepted implementation use mutex on setters/getters.

Reply:
Done.